models/Schematic_DataModel.php

```diff
@@ -49,12 +49,45 @@ public static function fromYaml($yaml, $overrideYaml)
         $data = Yaml::parse($yaml);
         $overrideData = Yaml::parse($overrideYaml);
         if ($overrideData != null) {
-            $mergedData = array_replace_recursive($data, $overrideData);
-        } else {
-            $mergedData = $data;
+            $data = array_replace_recursive($data, $overrideData);
         }
+        $data = static::replaceEnvVariables($data);
 
-        return $mergedData === null ? null : new static($mergedData);
+        return $data === null ? null : new static($data);
+    }
+
+    /**
+     * Replace placeholders with enviroment variables in array.
+     *
+     * Placeholders start with % and end with %. This will be replaced by the
+     * environment variable with the name SCHEMATIC_{PLACEHOLDER}. If the
+     * environment variable is not set an exception will be thrown.
+     *
+     * @param array $yaml
+     *
+     * @return array
+     */
+    public static function replaceEnvVariables($yaml)
+    {
+        array_walk_recursive($yaml, function(&$v) {
```

Comment:
Yes, I agree it would be faster, but requires more code changes and I don't think that is necessary at the moment.